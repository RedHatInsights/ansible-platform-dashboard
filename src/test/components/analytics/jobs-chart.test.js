import React from 'react';
import { render, waitFor, screen } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import promiseMiddleware from 'redux-promise-middleware';
import { MemoryRouter } from 'react-router-dom';
import notificationsMiddleware from '@redhat-cloud-services/frontend-components-notifications/notificationsMiddleware';
import analyticsReducer, { analyticsInitialState } from '../../../redux/reducers/analytics-reducer';
import JobsChart from '../../../components/analytics/jobs-chart';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { applyReducerHash, ReducerRegistry } from '@redhat-cloud-services/frontend-components-utilities/ReducerRegistry';

const ComponentWrapper = ({ store, initialEntries = [ '/ansible-dashboard' ], children }) => (
  <Provider store={ store } >
    <MemoryRouter initialEntries={ initialEntries }>
      <IntlProvider locale="en">
        { children }
      </IntlProvider>
    </MemoryRouter>
  </Provider>
);

describe('<JobsChart />', () => {
  let initialProps;
  let initialState;
  const middlewares = [thunk, promiseMiddleware, notificationsMiddleware()];
  let mockStore;

  beforeEach(() => {
    global.insights = {
      chrome: {
        identifyApp: jest.fn(),
        init: jest.fn(),
        auth: {
          getUser: () => Promise.resolve(true)
        }
      }
    };
    initialProps = {};
    mockStore = configureStore(middlewares);
    initialState = {
      i18nReducer: {
        formatMessage: ({ defaultMessage }) => defaultMessage
      },
      analyticsReducer: { ...analyticsInitialState, isLoading: false }
    };
  });

  it('should render correctly', async ()=> {
    const store = mockStore(initialState);
    const registry = new ReducerRegistry({}, [ thunk, promiseMiddleware ]);
    registry.register({ analyticsReducer: applyReducerHash(analyticsReducer, analyticsInitialState) });

    const { asFragment } = render(<ComponentWrapper store={ store }><JobsChart { ...initialProps } /></ComponentWrapper>);

    await waitFor(() => expect(() => screen.getByLabelText('Contents')).toThrow());

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly in loading state', () => {
    const store = mockStore(initialState);
    const { asFragment } = render(<ComponentWrapper store={ store }><JobsChart { ...initialProps } isLoading={ true } /></ComponentWrapper>);

    expect(asFragment()).toMatchSnapshot();
  });
});
