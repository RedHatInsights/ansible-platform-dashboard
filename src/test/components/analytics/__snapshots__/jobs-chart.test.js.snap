// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<JobsChart /> should render correctly 1`] = `
<IntlProvider
  defaultFormats={Object {}}
  defaultLocale="en"
  formats={Object {}}
  locale="en"
  messages={Object {}}
  onError={[Function]}
  textComponent={Symbol(react.fragment)}
>
  <Provider
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <MemoryRouter
      initialEntries={
        Array [
          "/ansible-dashboard",
        ]
      }
    >
      <IntlProvider
        defaultFormats={Object {}}
        defaultLocale="en"
        formats={Object {}}
        locale="en"
        messages={Object {}}
        onError={[Function]}
        textComponent={Symbol(react.fragment)}
      >
        <JobsChart />
      </IntlProvider>
    </MemoryRouter>
  </Provider>
</IntlProvider>
`;

exports[`<JobsChart /> should render correctly in loading state 1`] = `null`;
